contracts/account/extensions/AccountERC7579.sol

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {PackedUserOperation, IAccountExecute} from "@openzeppelin/contracts/interfaces/draft-IERC4337.sol";
+import {IERC1271} from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import {IERC7579Module, IERC7579Validator, IERC7579Execution, IERC7579AccountConfig, IERC7579ModuleConfig, MODULE_TYPE_VALIDATOR, MODULE_TYPE_EXECUTOR, MODULE_TYPE_FALLBACK} from "@openzeppelin/contracts/interfaces/draft-IERC7579.sol";
+import {ERC7579Utils, Mode, CallType, ExecType} from "@openzeppelin/contracts/account/utils/draft-ERC7579Utils.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {Packing} from "@openzeppelin/contracts/utils/Packing.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {ERC7739Signer} from "../../utils/cryptography/ERC7739Signer.sol";
+import {AccountCore} from "../AccountCore.sol";
+
+/**
+ * @dev Extension of {AccountCore} that implements support for ERC-7579 modules.
+ *
+ * To comply with the ERC-1271 support requirement, this contract implements {ERC7739Signer} as an
+ * opinionated layer to avoid signature replayability across accounts controlled by the same key.
+ *
+ * This contract does not implement validation logic for user operations since these functionality
+ * is often delegated to self-contained validation modules. Developers must install a validator module
+ * upon initialization (or any other mechanism to enable execution from the account):
+ *
+ * ```solidity
+ * contract MyAccountERC7579 is AccountERC7579, Initializable {
+ *     constructor() EIP712("MyAccountRSA", "1") {}
+ *
+ *   function initializeAccount(address validator, bytes calldata validatorData) public initializer {
```

Comment:
It's worth noting that this is vulnerable to frontrunning if used as a 7702 account.

I wonder if it might be worth defining a special-purpose `accountInitializer` modifier that is "safe for accounts" whether used as 7702 templates or standalone contracts.

Reply:
We can warn about this in concrete, I'm not sure if ERC-7579 will be default to enable 7702 accounts

Reply:
see https://github.com/OpenZeppelin/openzeppelin-community-contracts/blob/e26066dbcb0f160307ffacc9969389673710ab0b/contracts/mocks/account/AccountERC7702WithModulesMock.sol